presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreenState.kt

```diff
@@ -23,6 +23,6 @@ data class EventScreenState(
     val endDate: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault()),
     val isStartDateTime: Boolean = false,
     val shouldOpenDropdown: Boolean = false,
-    val alarmItem: AlarmItem = AlarmItem(ZonedDateTime.now(), null),
+    val alarmItem: AlarmItem = AlarmItem(ZonedDateTime.now(), ""),
```

Comment:
@philipplackner @MKiperszmid I am just wondering if the AlarmItem should be null or not.

Reply:
I'm not understanding the use of this

You should just save a ZonedDateTime, nothing else

Then, once the alarm get's triggered, you should get the event/task/reminder, you should get all info from the repository (db) based on the id

Reply:
@MKiperszmid if we save the just the ZoneDateTime and nothing else. For example the alarm is set to go off at 3pm. It then triggers a query to the DB to get the title, description and show the notification. However, it will also need the ID of the event to know which event it needs to query for. Am I missing something here?

Reply:
There's no need for the alarm to query the DB. The flow is like the following:
1. User saves agenda item with reminder type set to something like 30min before
2. That translates for example to a remindAt timestamp which is 30min before the agenda item time
3. You use AlarmManager to schedule an action with that remindAt timestamp. That requires passing an Intent to which you can pass all the info that is needed on the notification (title, description, etc. - in the end your `AlarmItem`). From that on it will stay scheduled
4. At the scheduled timestamp, AlarmManager will send a broadcast to your app which you can receive in a BroadcastReceiver
5. In that receiver, you get the passed data from step 3 and you can show the notification with that
6. When the device reboots, all alarms are deleted again. Therefore, after device reboot, you need to go through all future agenda items and reset their alarms (as I mentioned in my earlier message)